src/lib.rs

```diff
@@ -318,18 +318,37 @@ impl GlobWalkerBuilder {
 
     /// Finalize and build a `GlobWalker` instance.
     pub fn build(self) -> Result<GlobWalker, GlobError> {
-        let mut builder = OverrideBuilder::new(self.root);
+        let mut builder = OverrideBuilder::new(&self.root);
+        let mut parent_paths_builder = OverrideBuilder::new(&self.root);
 
         builder
             .case_insensitive(self.case_insensitive)
             .map_err(GlobError)?;
 
+        parent_paths_builder
+            .case_insensitive(self.case_insensitive)
+            .map_err(GlobError)?;
+
         for pattern in self.patterns {
             builder.add(pattern.as_ref()).map_err(GlobError)?;
+            let pattern_path = PathBuf::from(pattern);
+
+            for parent in pattern_path.ancestors().skip(1) {
```

Comment:
Could you comment on why `skip(1)` is needed?

Reply:
Because the first path yielded by the Ancestors iterator is the full path and we only care about the first ancestor.